tokio/src/sync/batch_semaphore.rs

```diff
@@ -41,15 +41,21 @@ struct Waitlist {
     closed: bool,
 }
 
-/// Error returned by `Semaphore::try_acquire`.
+/// Error returned from the `Semaphore::try_acquire` function.
 #[derive(Debug)]
-pub(crate) enum TryAcquireError {
+pub enum TryAcquireError {
+    /// The semaphore has been closed and cannot issue new permits
```

Comment:
Might be worth linking to the `Semaphore::close` method here?
```suggestion
    /// The semaphore has been [closed] and cannot issue new permits.
    ///
    /// [closed]: crate::sync::Semaphore::close
```